packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -659,6 +660,29 @@ export function getInitialSearchIndexPipeline(name: string) {
   ];
 }
 
+export function getInitialVectorSearchIndexPipelineText(name: string) {
+  return `[
+  {
+    $vectorSearch: {
+      // Name of the Atlas Vector Search index to use.
+      index: "${name}",
```

Comment:
```suggestion
      index: ${JSON.stringify(name)},
```

Hopefully not something where characters that need to be escaped show up frequently, but probably also no reason not to make this bit resilient against that